### PR TITLE
Add runner settings to IntelliJ remote debug config

### PR DESCRIPTION
**Pull request by adnapibar**
Signed-off-by: Rabi Panda <adnapibar@gmail.com>

### Description
The runner settings is automatically added if missing, thus adding this section to the debug config.

### Issues Resolved
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@adnapibar I think we are IDE neutral and should restrain from including any IDE specific files along with repository (f.e. I use Eclipse and/or VSCode)

**Comment by kotwanikunal** [comment]
> @adnapibar I think we are IDE neutral and should restrain from including any IDE specific files along with repository (f.e. I use Eclipse and/or VSCode), @dblock @andrross what do you think guys?

I think the pre-exiting `.idea` folder already has IntelliJ specific configuration.

**Comment by adnapibar** [comment]
> I think the pre-exiting `.idea` folder already has IntelliJ specific configuration.

Yes it already exists and we have [documented it in our testing guide](https://github.com/opensearch-project/OpenSearch/blob/main/TESTING.md#launching-and-debugging-from-an-ide). It's very helpful, harmless and doesn't force anyone to use IntelliJ though.

**Comment by reta** [comment]
> > I think the pre-exiting `.idea` folder already has IntelliJ specific configuration.
> 
> Yes it already exists and we have [documented it in our testing guide](https://github.com/opensearch-project/OpenSearch/blob/main/TESTING.md#launching-and-debugging-from-an-ide). It's very helpful, harmless and doesn't force anyone to use IntelliJ though.

Sure, but we should be open to other IDEs, if people would like to submit configuration / settings. In all cases - it is  very helpful, harmless and doesn't force anyone to use IDE of one's choice.

**Comment by adnapibar** [comment]
> > > I think the pre-exiting `.idea` folder already has IntelliJ specific configuration.
> > 
> > 
> > Yes it already exists and we have [documented it in our testing guide](https://github.com/opensearch-project/OpenSearch/blob/main/TESTING.md#launching-and-debugging-from-an-ide). It's very helpful, harmless and doesn't force anyone to use IntelliJ though.
> 
> Sure, but we should be open to other IDEs, if people would like to submit configuration / settings. In all cases - it is very helpful, harmless and doesn't force anyone to use IDE of one's choice.

I would be okay with that. Regardless, the point of this issue is that we already have this config in our repo and it has an issue.  If we are going to be IDE neutral, we should remove the IntelliJ configs and update our documentation?

**Comment by kotwanikunal** [comment]
@adnapibar Do you want to update the PR with configs for other IDEs?

**Comment by adnapibar** [comment]
> @adnapibar Do you want to update the PR with configs for other IDEs?

No plans for that now and I'm not well versed with other IDEs

File: CHANGELOG.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added missing no-jdk distributions ([#4722](https://github.com/opensearch-project/OpenSearch/pull/4722))
 - Add dev help in gradle check CI failures ([4872](https://github.com/opensearch-project/OpenSearch/pull/4872))
 - Copy `build.sh` over from opensearch-build ([#4887](https://github.com/opensearch-project/OpenSearch/pull/4887))
+- Add runner settings to IntelliJ remote debug config ([#4896](https://github.com/opensearch-project/OpenSearch/pull/4896))
```
Comment (andrross) [inline CHANGELOG.md:44]: I think we can skip the changelog since this is a dev-only change

**Pull request closed without merging**